### DISCUSSION TITLE
How do I create an interface from useMachine's state and send values

### DISCUSSION THREAD

**Discussion by OlaoluwaM** [statelyai/xstate#2490] [Q&A] [answered]
I am using a state machine to keep track of the global state along with React context. I am also doing all that with TS. I am trying to create an interface for my default context value. This interface will comprise the state node itself and the send dispatch function. 

More precisely, I mean the `state` and `send` values from `useMachine` ` [state, send] = useMachine(someMachine)`. The interface would look something like this

`interface ContextInterface { state: SomeType, sendEvent: SomeType }` Do you have any recommendations for what `SomeType` should be? Currently, I am using `any` because `unknown` would lead to too much type narrowing.

**Comment by Andarist** [comment, marked as answer]
I'm not entirely sure if that's what you are looking for but something like this can be used:
```ts
import { StateFrom, EventFrom } from 'xstate'

interface ContextInterface {
  state: StateFrom<typeof someMachine>;
  sendEvent: (event: EvenFrom<typeof someMachine>) => void
}
```

**Reply by OlaoluwaM** [reply]
Thank you so much! That worked. It would be very helpful if the docs made mention to these generics under the TS section (I can help with that BTW). Otherwise, y'all are doing a great job 🚀

**Reply by Andarist** [reply]
PRs are always welcome - feel free to prepare one ❤️
